scroll_overlay/android/app/src/main/java/io/flutter/scroll_overlay/ScrollOverlayActivity.java

```diff
@@ -3,151 +3,191 @@
-            flutterView.destroy();
-        }
+        FlutterEngineCache.getInstance().remove(engineId);
+        flutterEngine.destroy();
+        flutterEngine = null;
         super.onDestroy();
     }
 
+
     @Override
     protected void onPause() {
         super.onPause();
-        flutterView.onPause();
+        flutterEngine.getLifecycleChannel().appIsInactive();
     }
 
     @Override
     protected void onPostResume() {
         super.onPostResume();
-        flutterView.onPostResume();
+        flutterEngine.getLifecycleChannel().appIsResumed();
     }
 
     @Override
     public void onBackPressed() {
-        if (flutterView != null) {
-            flutterView.popRoute();
-            return;
-        }
+        flutterEngine.getNavigationChannel().popRoute();
         super.onBackPressed();
     }
 
     @Override
     protected void onStop() {
-        if (flutterView != null) {
-            flutterView.onStop();
-        }
+        flutterEngine.getLifecycleChannel().appIsPaused();
         super.onStop();
     }
 
-    @Override
-    protected void onStart() {
-        super.onStart();
-        if (flutterView != null) {
-            flutterView.onStart();
-        }
-    }
-
-    private class OverlayAdapter extends BaseAdapter {
+    private class OverlayAdapter extends RecyclerView.Adapter<OverlayAdapter.OverlayViewHolder> {
+        @NonNull
         @Override
-        public int getCount() {
-            return Integer.MAX_VALUE;
+        public OverlayViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
+            LayoutInflater inflater = LayoutInflater.from(ScrollOverlayActivity.this);
+            View view = inflater.inflate(R.layout.row, parent, false);
+            return new OverlayViewHolder(view);
         }
 
         @Override
-        public Object getItem(int position) {
-            return position;
+        public void onBindViewHolder(@NonNull OverlayViewHolder holder, int position) {
+            holder.bind(position);
         }
 
         @Override
-        public long getItemId(int position) {
-            return position;
+        public int getItemCount() {
+            return Integer.MAX_VALUE;
         }
 
-        @Override
-        public View getView(int position, View convertView, ViewGroup parent) {
-            if (convertView == null) {
-                TextView textView = new TextView(ScrollOverlayActivity.this);
+        class OverlayViewHolder extends RecyclerView.ViewHolder {
+            public OverlayViewHolder(@NonNull View itemView) {
+                super(itemView);
+            }
+
+            void bind(int position) {
+                int color = OVERLAY_COLORS[position % OVERLAY_COLORS.length];
+                itemView.setBackground(new ColorDrawable(color));
+                final int height = (int) TypedValue
+                        .applyDimension(TypedValue.COMPLEX_UNIT_DIP, 40 + position, getResources().getDisplayMetrics());
```

Comment:
`40` was already in the code, but would you mind creating a constant for it that auto explains what it controls?

Reply:
done, both in dart and native